### PR TITLE
data-platform-minimal - support web_server_network_access_control

### PR DESCRIPTION
**Pull request by kthhrv**
enable setting of composer web server network access control

### CODE REVIEW

File: blueprints/data-solutions/data-platform-minimal/variables.tf [hunk 1]
```diff
@@ -24,6 +24,16 @@ variable "composer_config" {
       env_variables            = optional(map(string), {})
       image_version            = optional(string, "composer-2-airflow-2")
     }), {})
+    web_server_network_access_ip_ranges = optional(list(map(string)), [
```
Comment (juliocc) [inline blueprints/data-solutions/data-platform-minimal/variables.tf:27]: - I'd keep the original name from the provider `web_server_network_access_control` or maybe even `web_server_access_control`.
- For the type I think `optional(map(string))` without any default a better option. You can use it with something like  
  ```terraform
  web_server_access_control = { 
    "10.0.0.0/8" = "Allow 10/8"
    "192.168.0.0/16" = "Allow 192.168/16"
  } 
  ```

**Review by juliocc — CHANGES_REQUESTED** [review]
Looks great, thanks!

Check my suggestions below.

File: blueprints/data-solutions/data-platform-minimal/variables.tf (hunk 1 above)
Comment (kthhrv) [reply blueprints/data-solutions/data-platform-minimal/variables.tf:27]: cool
- I'd used a list of maps to support multiple ranges (which composer supports)
```
web_server_network_access_ip_ranges = [
  {
    description = "Range 1"
    value = "1.2.3.4/32"
  },
  {
    description = "Range 2"
    value = "4.5.6.7/32"
  }
]
```
- I provided a default that matches the actually composer default so that if you remove your custom ranges it reverts to the defaults, I'll double check that it doesn't just revert to them anyway.

Finishing up some more testing of the edge cases

Comment (kthhrv) [reply blueprints/data-solutions/data-platform-minimal/variables.tf:27]: Ah, i'll try a single map, thanks

the default for a composer web server is `public` so replicating that for the default is ok?

can you explain the remaining linting error? couldn't make heads or tails of the failure message, cheers!

Comment (juliocc) [reply blueprints/data-solutions/data-platform-minimal/variables.tf:27]: > Ah, i'll try a single map, thanks. so the for_each will stay the same?

The for_each yes, but you'll probably have to change the content of the dynamic block

> the default for a composer web server is public so replicating that for the default is ok?

I'd rather no. Just leave it without default.

> can you explain the remaining linting error? couldn't make heads or tails of the failure message, cheers!

You have to update the variables table. We have a script to do that, check our [contributing guide](https://github.com/GoogleCloudPlatform/cloud-foundation-fabric/blob/master/CONTRIBUTING.md#quick-developer-workflow).

**Comment by kthhrv** [comment]
still need to refactor the for_each content

**Review by juliocc — APPROVED** [review]
@kthhrv are you ready to merge this? This looks goods to me.

**Comment by kthhrv** [comment]
I've broken my local dev/test environment by rebasing to latest master, would like to get that working again to finishing my testing of this change before merging, thanks

**Comment by kthhrv** [comment]
here is the output of the 3rd apply

> ~/ws/fast-03-data-platform-prod$ tf apply
module.data-platform.module.cur-project.google_project.project[0]: Refreshing state... [id=projects/tosbx-prod-dt-cur]
module.data-platform.module.processing-project.google_project.project[0]: Refreshing state... [id=projects/tosbx-prod-dt-prc]
module.data-platform.module.land-project.google_project.project[0]: Refreshing state... [id=projects/tosbx-prod-dt-lnd]
module.data-platform.module.common-project.google_project.project[0]: Refreshing state... [id=projects/tosbx-prod-dt-cmn]
module.data-platform.module.common-project.google_project_service.project_services["serviceusage.googleapis.com"]: Refreshing state... [id=tosbx-prod-dt-cmn/serviceusage.googleapis.com]
module.data-platform.module.common-project.google_project_service.project_services["dlp.googleapis.com"]: Refreshing state... [id=tosbx-prod-dt-cmn/dlp.googleapis.com]
module.data-platform.module.common-project.google_project_service.project_services["stackdriver.googleapis.com"]: Refreshing state... [id=tosbx-prod-dt-cmn/stackdriver.googleapis.com]
module.data-platform.module.common-project.google_project_service.project_services["iam.googleapis.com"]: Refreshing state... [id=tosbx-prod-dt-cmn/iam.googleapis.com]
module.data-platform.module.common-project.google_project_service.project_services["datacatalog.googleapis.com"]: Refreshing state... [id=tosbx-prod-dt-cmn/datacatalog.googleapis.com]
module.data-platform.module.common-project.google_project_service.project_services["cloudresourcemanager.googleapis.com"]: Refreshing state... [id=tosbx-prod-dt-cmn/cloudresourcemanager.googleapis.com]
module.data-platform.module.processing-project.google_project_service.project_services["storage-component.googleapis.com"]: Refreshing state... [id=tosbx-prod-dt-prc/storage-component.googleapis.com]
module.data-platform.module.processing-project.google_project_service.project_services["stackdriver.googleapis.com"]: Refreshing state... [id=tosbx-prod-dt-prc/stackdriver.googleapis.com]
module.data-platform.module.processing-project.google_project_service.project_services["composer.googleapis.com"]: Refreshing state... [id=tosbx-prod-dt-prc/composer.googleapis.com]
module.data-platform.module.processing-project.google_project_service.project_services["bigquerystorage.googleapis.com"]: Refreshing state... [id=tosbx-prod-dt-prc/bigquerystorage.googleapis.com]
module.data-platform.module.processing-project.google_project_service.project_services["serviceusage.googleapis.com"]: Refreshing state... [id=tosbx-prod-dt-prc/serviceusage.googleapis.com]
module.data-platform.module.processing-project.google_project_service.project_services["cloudkms.googleapis.com"]: Refreshing state... [id=tosbx-prod-dt-prc/cloudkms.googleapis.com]
module.data-platform.module.processing-project.google_project_service.project_services["servicenetworking.googleapis.com"]: Refreshing state... [id=tosbx-prod-dt-prc/servicenetworking.googleapis.com]
module.data-platform.module.processing-project.google_project_service.project_services["bigquery.googleapis.com"]: Refreshing state... [id=tosbx-prod-dt-prc/bigquery.googleapis.com]
module.data-platform.module.processing-project.google_project_service.project_services["iam.googleapis.com"]: Refreshing state... [id=tosbx-prod-dt-prc/iam.googleapis.com]
module.data-platform.module.processing-project.google_project_service.project_services["bigqueryreservation.googleapis.com"]: Refreshing state... [id=tosbx-prod-dt-prc/bigqueryreservation.googleapis.com]
module.data-platform.module.processing-project.google_project_service.project_services["cloudresourcemanager.googleapis.com"]: Refreshing state... [id=tosbx-prod-dt-prc/cloudresourcemanager.googleapis.com]
module.data-platform.module.processing-project.google_project_service.project_services["dataproc.googleapis.com"]: Refreshing state... [id=tosbx-prod-dt-prc/dataproc.googleapis.com]
module.data-platform.module.processing-project.google_project_service.project_services["container.googleapis.com"]: Refreshing state... [id=tosbx-prod-dt-prc/container.googleapis.com]
module.data-platform.module.processing-project.google_project_service.project_services["dataflow.googleapis.com"]: Refreshing state... [id=tosbx-prod-dt-prc/dataflow.googleapis.com]
module.data-platform.module.processing-project.google_compute_shared_vpc_service_project.shared_vpc_service[0]: Refreshing state... [id=tosbx-prod-net-spoke-0/tosbx-prod-dt-prc]
module.data-platform.module.cur-project.google_project_service.project_services["iam.googleapis.com"]: Refreshing state... [id=tosbx-prod-dt-cur/iam.googleapis.com]
module.data-platform.module.processing-project.google_project_service.project_services["compute.googleapis.com"]: Refreshing state... [id=tosbx-prod-dt-prc/compute.googleapis.com]
module.data-platform.module.cur-project.google_project_service.project_services["serviceusage.googleapis.com"]: Refreshing state... [id=tosbx-prod-dt-cur/serviceusage.googleapis.com]
module.data-platform.module.cur-project.google_project_service.project_services["compute.googleapis.com"]: Refreshing state... [id=tosbx-prod-dt-cur/compute.googleapis.com]
module.data-platform.module.cur-project.google_project_service.project_services["cloudresourcemanager.googleapis.com"]: Refreshing state... [id=tosbx-prod-dt-cur/cloudresourcemanager.googleapis.com]
module.data-platform.module.cur-project.google_project_service.project_services["bigqueryreservation.googleapis.com"]: Refreshing state... [id=tosbx-prod-dt-cur/bigqueryreservation.googleapis.com]
module.data-platform.module.processing-project.google_project_service.project_services["storage.googleapis.com"]: Refreshing state... [id=tosbx-prod-dt-prc/storage.googleapis.com]
module.data-platform.module.cur-project.google_project_service.project_services["stackdriver.googleapis.com"]: Refreshing state... [id=tosbx-prod-dt-cur/stackdriver.googleapis.com]
module.data-platform.module.cur-project.google_project_service.project_services["bigquerystorage.googleapis.com"]: Refreshing state... [id=tosbx-prod-dt-cur/bigquerystorage.googleapis.com]
module.data-platform.module.cur-project.google_project_service.project_services["storage.googleapis.com"]: Refreshing state... [id=tosbx-prod-dt-cur/storage.googleapis.com]
module.data-platform.module.cur-project.google_project_service.project_services["bigquery.googleapis.com"]: Refreshing state... [id=tosbx-prod-dt-cur/bigquery.googleapis.com]
module.data-platform.module.cur-project.google_project_service.project_services["cloudkms.googleapis.com"]: Refreshing state... [id=tosbx-prod-dt-cur/cloudkms.googleapis.com]
module.data-platform.module.cur-project.google_project_service.project_services["storage-component.googleapis.com"]: Refreshing state... [id=tosbx-prod-dt-cur/storage-component.googleapis.com]
module.data-platform.module.cur-project.google_project_service.project_services["servicenetworking.googleapis.com"]: Refreshing state... [id=tosbx-prod-dt-cur/servicenetworking.googleapis.com]
module.data-platform.module.land-project.google_project_service.project_services["bigquery.googleapis.com"]: Refreshing state... [id=tosbx-prod-dt-lnd/bigquery.googleapis.com]
module.data-platform.module.land-project.google_project_service.project_services["storage.googleapis.com"]: Refreshing state... [id=tosbx-prod-dt-lnd/storage.googleapis.com]
module.data-platform.module.land-project.google_project_service.project_services["serviceusage.googleapis.com"]: Refreshing state... [id=tosbx-prod-dt-lnd/serviceusage.googleapis.com]
module.data-platform.module.land-project.google_project_service.project_services["stackdriver.googleapis.com"]: Refreshing state... [id=tosbx-prod-dt-lnd/stackdriver.googleapis.com]
module.data-platform.module.land-project.google_project_service.project_services["storage-component.googleapis.com"]: Refreshing state... [id=tosbx-prod-dt-lnd/storage-component.googleapis.com]
module.data-platform.module.land-project.google_project_service.project_services["bigqueryreservation.googleapis.com"]: Refreshing state... [id=tosbx-prod-dt-lnd/bigqueryreservation.googleapis.com]
module.data-platform.module.land-project.google_project_service.project_services["cloudkms.googleapis.com"]: Refreshing state... [id=tosbx-prod-dt-lnd/cloudkms.googleapis.com]
module.data-platform.module.land-project.google_project_service.project_services["bigquerystorage.googleapis.com"]: Refreshing state... [id=tosbx-prod-dt-lnd/bigquerystorage.googleapis.com]
module.data-platform.module.land-project.google_project_service.project_services["iam.googleapis.com"]: Refreshing state... [id=tosbx-prod-dt-lnd/iam.googleapis.com]
module.data-platform.module.land-project.google_project_service.project_services["cloudresourcemanager.googleapis.com"]: Refreshing state... [id=tosbx-prod-dt-lnd/cloudresourcemanager.googleapis.com]
module.data-platform.module.processing-project.data.google_storage_project_service_account.gcs_sa[0]: Reading...
module.data-platform.module.processing-project.data.google_bigquery_default_service_account.bq_sa[0]: Reading...
module.data-platform.module.cur-project.google_project_service_identity.servicenetworking[0]: Refreshing state... [id=projects/tosbx-prod-dt-cur/services/servicenetworking.googleapis.com]
module.data-platform.module.processing-project.google_project_service_identity.servicenetworking[0]: Refreshing state... [id=projects/tosbx-prod-dt-prc/services/servicenetworking.googleapis.com]
module.data-platform.module.cur-project.data.google_storage_project_service_account.gcs_sa[0]: Reading...
module.data-platform.module.cur-project.data.google_bigquery_default_service_account.bq_sa[0]: Reading...
module.data-platform.module.common-datacatalog.google_data_catalog_taxonomy.default: Refreshing state... [id=projects/tosbx-prod-dt-cmn/locations/eu/taxonomies/8690897363005152116]
module.data-platform.module.processing-project.google_project_iam_member.servicenetworking[0]: Refreshing state... [id=tosbx-prod-dt-prc/roles/servicenetworking.serviceAgent/serviceAccount:service-922934110375@service-networking.iam.gserviceaccount.com]
module.data-platform.module.cur-project.google_project_iam_member.servicenetworking[0]: Refreshing state... [id=tosbx-prod-dt-cur/roles/servicenetworking.serviceAgent/serviceAccount:service-78517993489@service-networking.iam.gserviceaccount.com]
module.data-platform.module.cur-project.data.google_storage_project_service_account.gcs_sa[0]: Read complete after 0s [id=service-78517993489@gs-project-accounts.iam.gserviceaccount.com]
module.data-platform.module.processing-project.data.google_storage_project_service_account.gcs_sa[0]: Read complete after 0s [id=service-922934110375@gs-project-accounts.iam.gserviceaccount.com]
module.data-platform.module.common-datacatalog.google_data_catalog_policy_tag.default["3_Confidential"]: Refreshing state... [id=projects/tosbx-prod-dt-cmn/locations/eu/taxonomies/8690897363005152116/policyTags/8510401906131669078]
module.data-platform.module.common-datacatalog.google_data_catalog_policy_tag.default["2_Private"]: Refreshing state... [id=projects/tosbx-prod-dt-cmn/locations/eu/taxonomies/8690897363005152116/policyTags/1807636667494686715]
module.data-platform.module.common-datacatalog.google_data_catalog_policy_tag.default["1_Sensitive"]: Refreshing state... [id=projects/tosbx-prod-dt-cmn/locations/eu/taxonomies/8690897363005152116/policyTags/2935122188501596687]
module.data-platform.module.processing-project.data.google_bigquery_default_service_account.bq_sa[0]: Read complete after 0s [id=bq-922934110375@bigquery-encryption.iam.gserviceaccount.com]
module.data-platform.module.processing-project.google_project_iam_member.shared_vpc_host_robots["roles/compute.networkUser:cloudservices"]: Refreshing state... [id=tosbx-prod-net-spoke-0/roles/compute.networkUser/serviceAccount:922934110375@cloudservices.gserviceaccount.com]
module.data-platform.module.processing-project.google_project_iam_member.shared_vpc_host_robots["roles/compute.networkUser:container-engine"]: Refreshing state... [id=tosbx-prod-net-spoke-0/roles/compute.networkUser/serviceAccount:service-922934110375@container-engine-robot.iam.gserviceaccount.com]
module.data-platform.module.processing-project.google_project_iam_member.shared_vpc_host_robots["roles/container.hostServiceAgentUser:container-engine"]: Refreshing state... [id=tosbx-prod-net-spoke-0/roles/container.hostServiceAgentUser/serviceAccount:service-922934110375@container-engine-robot.iam.gserviceaccount.com]
module.data-platform.module.processing-project.google_project_iam_member.shared_vpc_host_robots["roles/compute.networkUser:compute"]: Refreshing state... [id=tosbx-prod-net-spoke-0/roles/compute.networkUser/serviceAccount:service-922934110375@compute-system.iam.gserviceaccount.com]
module.data-platform.module.processing-project.google_project_iam_member.shared_vpc_host_robots["roles/compute.networkUser:dataproc"]: Refreshing state... [id=tosbx-prod-net-spoke-0/roles/compute.networkUser/serviceAccount:service-922934110375@dataproc-accounts.iam.gserviceaccount.com]
module.data-platform.module.cur-project.data.google_bigquery_default_service_account.bq_sa[0]: Read complete after 0s [id=bq-78517993489@bigquery-encryption.iam.gserviceaccount.com]
module.data-platform.module.processing-project.google_project_iam_member.shared_vpc_host_robots["roles/compute.networkUser:dataflow"]: Refreshing state... [id=tosbx-prod-net-spoke-0/roles/compute.networkUser/serviceAccount:service-922934110375@dataflow-service-producer-prod.iam.gserviceaccount.com]
module.data-platform.module.processing-project.google_project_iam_member.shared_vpc_host_robots["roles/composer.sharedVpcAgent:composer"]: Refreshing state... [id=tosbx-prod-net-spoke-0/roles/composer.sharedVpcAgent/serviceAccount:service-922934110375@cloudcomposer-accounts.iam.gserviceaccount.com]
module.data-platform.module.cur-sa-0.google_service_account.service_account[0]: Refreshing state... [id=projects/tosbx-prod-dt-cur/serviceAccounts/tosbx-prod-dt-cur-sa-0@tosbx-prod-dt-cur.iam.gserviceaccount.com]
module.data-platform.module.cur-bq-0.google_bigquery_dataset.default: Refreshing state... [id=projects/tosbx-prod-dt-cur/datasets/tosbx_prod_dt_cur_bq_0]
module.data-platform.module.cur-cs-0.google_storage_bucket.bucket: Refreshing state... [id=tosbx-prod-dt-cur-cs-0]
module.data-platform.module.cur-sa-0.google_service_account_iam_binding.roles["roles/iam.serviceAccountTokenCreator"]: Refreshing state... [id=projects/tosbx-prod-dt-cur/serviceAccounts/tosbx-prod-dt-cur-sa-0@tosbx-prod-dt-cur.iam.gserviceaccount.com/roles/iam.serviceAccountTokenCreator]
module.data-platform.module.processing-sa-0.google_service_account.service_account[0]: Refreshing state... [id=projects/tosbx-prod-dt-prc/serviceAccounts/tosbx-prod-dt-prc-0@tosbx-prod-dt-prc.iam.gserviceaccount.com]
module.data-platform.module.processing-sa-cmp-0.google_service_account.service_account[0]: Refreshing state... [id=projects/tosbx-prod-dt-prc/serviceAccounts/tosbx-prod-dt-prc-cmp-0@tosbx-prod-dt-prc.iam.gserviceaccount.com]
module.data-platform.module.processing-cs-0.google_storage_bucket.bucket: Refreshing state... [id=tosbx-prod-dt-prc-cs-0]
module.data-platform.module.processing-temp-0.google_storage_bucket.bucket: Refreshing state... [id=tosbx-prod-dt-prc-tmp-0]
module.data-platform.module.processing-staging-0.google_storage_bucket.bucket: Refreshing state... [id=tosbx-prod-dt-prc-stg-0]
module.data-platform.module.processing-log-0.google_storage_bucket.bucket: Refreshing state... [id=tosbx-prod-dt-prc-log-0]
module.data-platform.module.cur-project.google_project_iam_binding.authoritative["roles/storage.objectViewer"]: Refreshing state... [id=tosbx-prod-dt-cur/roles/storage.objectViewer]
module.data-platform.module.cur-project.google_project_iam_binding.authoritative["roles/bigquery.dataOwner"]: Refreshing state... [id=tosbx-prod-dt-cur/roles/bigquery.dataOwner]
module.data-platform.module.cur-project.google_project_iam_binding.authoritative["roles/datacatalog.viewer"]: Refreshing state... [id=tosbx-prod-dt-cur/roles/datacatalog.viewer]
module.data-platform.module.cur-project.google_project_iam_binding.authoritative["roles/bigquery.dataViewer"]: Refreshing state... [id=tosbx-prod-dt-cur/roles/bigquery.dataViewer]
module.data-platform.module.cur-project.google_project_iam_binding.authoritative["roles/datacatalog.tagTemplateViewer"]: Refreshing state... [id=tosbx-prod-dt-cur/roles/datacatalog.tagTemplateViewer]
module.data-platform.module.cur-project.google_project_iam_binding.authoritative["roles/storage.objectAdmin"]: Refreshing state... [id=tosbx-prod-dt-cur/roles/storage.objectAdmin]
module.data-platform.module.cur-project.google_project_iam_binding.authoritative["roles/bigquery.jobUser"]: Refreshing state... [id=tosbx-prod-dt-cur/roles/bigquery.jobUser]
module.data-platform.module.common-project.google_project_iam_binding.authoritative["roles/datacatalog.admin"]: Refreshing state... [id=tosbx-prod-dt-cmn/roles/datacatalog.admin]
module.data-platform.module.common-project.google_project_iam_binding.authoritative["roles/dlp.admin"]: Refreshing state... [id=tosbx-prod-dt-cmn/roles/dlp.admin]
module.data-platform.module.common-project.google_project_iam_binding.authoritative["roles/dlp.reader"]: Refreshing state... [id=tosbx-prod-dt-cmn/roles/dlp.reader]
module.data-platform.module.common-project.google_project_iam_binding.authoritative["roles/dlp.estimatesAdmin"]: Refreshing state... [id=tosbx-prod-dt-cmn/roles/dlp.estimatesAdmin]
module.data-platform.module.common-project.google_project_iam_binding.authoritative["roles/datacatalog.categoryFineGrainedReader"]: Refreshing state... [id=tosbx-prod-dt-cmn/roles/datacatalog.categoryFineGrainedReader]
module.data-platform.module.common-project.google_project_iam_binding.authoritative["roles/dlp.user"]: Refreshing state... [id=tosbx-prod-dt-cmn/roles/dlp.user]
module.data-platform.module.common-project.google_project_iam_binding.authoritative["roles/datacatalog.viewer"]: Refreshing state... [id=tosbx-prod-dt-cmn/roles/datacatalog.viewer]
module.data-platform.module.land-project.data.google_storage_project_service_account.gcs_sa[0]: Reading...
module.data-platform.module.land-project.data.google_bigquery_default_service_account.bq_sa[0]: Reading...
module.data-platform.module.processing-sa-0.google_service_account_iam_binding.roles["roles/iam.serviceAccountUser"]: Refreshing state... [id=projects/tosbx-prod-dt-prc/serviceAccounts/tosbx-prod-dt-prc-0@tosbx-prod-dt-prc.iam.gserviceaccount.com/roles/iam.serviceAccountUser]
module.data-platform.module.processing-sa-0.google_service_account_iam_binding.roles["roles/iam.serviceAccountTokenCreator"]: Refreshing state... [id=projects/tosbx-prod-dt-prc/serviceAccounts/tosbx-prod-dt-prc-0@tosbx-prod-dt-prc.iam.gserviceaccount.com/roles/iam.serviceAccountTokenCreator]
module.data-platform.module.land-project.data.google_storage_project_service_account.gcs_sa[0]: Read complete after 0s [id=service-695011761490@gs-project-accounts.iam.gserviceaccount.com]
module.data-platform.module.processing-sa-cmp-0.google_service_account_iam_binding.roles["roles/iam.serviceAccountTokenCreator"]: Refreshing state... [id=projects/tosbx-prod-dt-prc/serviceAccounts/tosbx-prod-dt-prc-cmp-0@tosbx-prod-dt-prc.iam.gserviceaccount.com/roles/iam.serviceAccountTokenCreator]
module.data-platform.module.land-project.data.google_bigquery_default_service_account.bq_sa[0]: Read complete after 0s [id=bq-695011761490@bigquery-encryption.iam.gserviceaccount.com]
module.data-platform.module.processing-sa-cmp-0.google_service_account_iam_binding.roles["roles/iam.serviceAccountUser"]: Refreshing state... [id=projects/tosbx-prod-dt-prc/serviceAccounts/tosbx-prod-dt-prc-cmp-0@tosbx-prod-dt-prc.iam.gserviceaccount.com/roles/iam.serviceAccountUser]
module.data-platform.module.processing-project.google_project_iam_binding.authoritative["roles/composer.admin"]: Refreshing state... [id=tosbx-prod-dt-prc/roles/composer.admin]
module.data-platform.module.processing-project.google_project_iam_binding.authoritative["roles/iap.httpsResourceAccessor"]: Refreshing state... [id=tosbx-prod-dt-prc/roles/iap.httpsResourceAccessor]
module.data-platform.module.processing-project.google_project_iam_binding.authoritative["roles/dataflow.admin"]: Refreshing state... [id=tosbx-prod-dt-prc/roles/dataflow.admin]
module.data-platform.module.processing-project.google_project_iam_binding.authoritative["roles/storage.admin"]: Refreshing state... [id=tosbx-prod-dt-prc/roles/storage.admin]
module.data-platform.module.processing-project.google_project_iam_binding.authoritative["roles/serviceusage.serviceUsageConsumer"]: Refreshing state... [id=tosbx-prod-dt-prc/roles/serviceusage.serviceUsageConsumer]
module.data-platform.module.processing-project.google_project_iam_binding.authoritative["roles/composer.worker"]: Refreshing state... [id=tosbx-prod-dt-prc/roles/composer.worker]
module.data-platform.module.processing-project.google_project_iam_binding.authoritative["roles/dataflow.worker"]: Refreshing state... [id=tosbx-prod-dt-prc/roles/dataflow.worker]
module.data-platform.module.processing-project.google_project_iam_binding.authoritative["roles/dataproc.editor"]: Refreshing state... [id=tosbx-prod-dt-prc/roles/dataproc.editor]
module.data-platform.module.processing-project.google_project_iam_binding.authoritative["roles/bigquery.jobUser"]: Refreshing state... [id=tosbx-prod-dt-prc/roles/bigquery.jobUser]
module.data-platform.module.processing-project.google_project_iam_binding.authoritative["roles/composer.environmentAndStorageObjectAdmin"]: Refreshing state... [id=tosbx-prod-dt-prc/roles/composer.environmentAndStorageObjectAdmin]
module.data-platform.module.processing-project.google_project_iam_binding.authoritative["roles/dataproc.worker"]: Refreshing state... [id=tosbx-prod-dt-prc/roles/dataproc.worker]
module.data-platform.module.processing-project.google_project_iam_binding.authoritative["roles/iam.serviceAccountUser"]: Refreshing state... [id=tosbx-prod-dt-prc/roles/iam.serviceAccountUser]
module.data-platform.module.processing-project.google_project_iam_binding.authoritative["roles/composer.ServiceAgentV2Ext"]: Refreshing state... [id=tosbx-prod-dt-prc/roles/composer.ServiceAgentV2Ext]
module.data-platform.module.land-cs-0.google_storage_bucket.bucket: Refreshing state... [id=tosbx-prod-dt-lnd-cs-0]
module.data-platform.module.land-sa-0.google_service_account.service_account[0]: Refreshing state... [id=projects/tosbx-prod-dt-lnd/serviceAccounts/tosbx-prod-dt-lnd-sa-0@tosbx-prod-dt-lnd.iam.gserviceaccount.com]
module.data-platform.module.land-bq-0.google_bigquery_dataset.default: Refreshing state... [id=projects/tosbx-prod-dt-lnd/datasets/tosbx_prod_dt_lnd_bq_0]
module.data-platform.module.land-sa-0.google_service_account_iam_binding.roles["roles/iam.serviceAccountTokenCreator"]: Refreshing state... [id=projects/tosbx-prod-dt-lnd/serviceAccounts/tosbx-prod-dt-lnd-sa-0@tosbx-prod-dt-lnd.iam.gserviceaccount.com/roles/iam.serviceAccountTokenCreator]
module.data-platform.module.land-project.google_project_iam_binding.authoritative["roles/storage.objectAdmin"]: Refreshing state... [id=tosbx-prod-dt-lnd/roles/storage.objectAdmin]
module.data-platform.module.land-project.google_project_iam_binding.authoritative["roles/storage.objectCreator"]: Refreshing state... [id=tosbx-prod-dt-lnd/roles/storage.objectCreator]
module.data-platform.module.land-project.google_project_iam_binding.authoritative["roles/storage.objectViewer"]: Refreshing state... [id=tosbx-prod-dt-lnd/roles/storage.objectViewer]
module.data-platform.google_composer_environment.processing-cmp-0[0]: Refreshing state... [id=projects/tosbx-prod-dt-prc/locations/europe-west1/environments/tosbx-prod-dt-prc-cmp-0]

No changes. Your infrastructure matches the configuration.

Terraform has compared your real infrastructure against your configuration and found no differences, so
no changes are needed.
╷
│ Warning: Value for undeclared variable
│ 
│ The root module does not declare a variable named "service_accounts" but a value was found in file
│ "1-resman.auto.tfvars.json". If you meant to use this value, add a "variable" block to the
│ configuration.
│ 
│ To silence these warnings, use TF_VAR_... environment variables to provide certain "global" settings
│ to all configurations in your organization. To reduce the verbosity of these warnings, use the
│ -compact-warnings option.
╵
╷
│ Warning: Value for undeclared variable
│ 
│ The root module does not declare a variable named "locations" but a value was found in file
│ "globals.auto.tfvars.json". If you meant to use this value, add a "variable" block to the
│ configuration.
│ 
│ To silence these warnings, use TF_VAR_... environment variables to provide certain "global" settings
│ to all configurations in your organization. To reduce the verbosity of these warnings, use the
│ -compact-warnings option.
╵
╷
│ Warning: Values for undeclared variables
│ 
│ In addition to the other similar warnings shown, 10 other variable(s) defined without being declared.
╵

Apply complete! Resources: 0 added, 0 changed, 0 destroyed.

**Comment by kthhrv** [comment]
@juliocc I'm new to TF and GCP so please excuse my lack on knowledge :) BTW I'm just using a temporary Sandbox organisation to learn and develop our IoC before deploying in anger to our real GCP Org

**Comment by juliocc** [comment]
> @juliocc I'm new to TF and GCP so please excuse my lack of knowledge :) BTW I'm just using a temporary Sandbox organisation to learn and develop our IoC before deploying in anger to our real GCP Org

That output is fine. IMO the module and terraform are WAI.

I'll go ahead and merge like this.

**Comment by kthhrv** [comment]
Thanks! "WAI"?

**Comment by juliocc** [comment]
> Thanks! "WAI"?

lol Working as intended. Sorry, we suffer from RES Syndrome

> > "WAI"?
> 
> Works As Intended :-)

Exactly! We like our TLAs ;)

**Comment by kthhrv** [comment]
this looks like someone having the same issue with composer not reverting to default if CIDRs aren't supplied
https://discuss.hashicorp.com/t/unable-to-remove-resource-attribute-that-was-provisioned-with-terraform/24701
this is bug with the provider module?
